### PR TITLE
feat: smaller networks for tests

### PR DESCRIPTION
**Pull request by sash-a**
## What?
Make test run faster by using tiny networks
## Why?
Faster is better?

### CODE REVIEW

**Comment by sash-a** [comment]
Looks like it's ~1 minute faster :racehorse:

File: test/conftest.py [hunk 1]
```diff
@@ -12,37 +12,67 @@
...
 
 import pytest
 
+_ConfSingleValue: TypeAlias = bool | int | float
+ConfigValue: TypeAlias = _ConfSingleValue | List[_ConfSingleValue] | Dict[str, _ConfSingleValue]
+
 
 @pytest.fixture
-def fast_config() -> Dict[str, Dict[str, bool | int | float]]:
+def fast_config_modifications() -> Dict[str, ConfigValue]:
```
Comment (WiemKhlifi) [inline test/conftest.py]: A stupid question: why `fast_config_modifications` instead of  `fast_config` 🙄

File: test/conftest.py [hunk 2]
```diff
@@ -12,37 +12,67 @@
...
+
+    Args:
+        d: Dictionary to search through
+        replacements: The keys and values to replace
+    """
+
+    def _find_replace_recursive(current_dict: Dict[str, ConfigValue]) -> Dict[str, ConfigValue]:
+        """Helper function that recursively searches and replaces values."""
+        x = {}
+        for k, v in list(current_dict.items()):
```
Comment (WiemKhlifi) [inline test/conftest.py]: Few things I remember from my PS classes is that converting these to a list slows down your code a bit 👀 
```suggestion
        for k, v in current_dict.items():
```

File: test/conftest.py [hunk 3]
```diff
@@ -12,37 +12,67 @@
...
+            if isinstance(v, dict):
+                v = _find_replace_recursive(v)
+            elif k in replacements:
+                v = replacements[k]
+
+            x[k] = v
+
+        return x
```
Comment (WiemKhlifi) [inline test/conftest.py]: Can't we just replace elements in the `current_dict`? ( tested  & it works)
```suggestion
            if isinstance(v, dict):
                current_dict[k] = _find_replace_recursive(v)
            elif k in replacements:
                current_dict[k] = replacements[k]

        return current_dict
```

File: test/integration_test.py [hunk 4]
```diff
@@ -19,6 +19,8 @@
 from hydra import compose, initialize
 from omegaconf import DictConfig, OmegaConf
 
+from test.conftest import find_replace
```
Comment (WiemKhlifi) [inline test/integration_test.py]: I'm not sure if this is good practice to import from `conftest` can't we just move this function into the `test/integration_test.py` or create a file for helper fcts?

**Review by WiemKhlifi — CHANGES_REQUESTED** [review]
Thanks @sash-a few suggestions but all LGTM 🥳

**Review by RuanJohn — APPROVED** [review]
Thanks for this! :raised_hands:

File: test/integration_test.py (hunk 4 above)
Comment (sash-a) [reply test/integration_test.py]: Ye agreed, it felt a bit weird to me, I'll move it :smile:

File: test/conftest.py [hunk 5]
```diff
@@ -12,37 +12,67 @@
...
+        # sac
+        "explore_steps": 1,
+        "epochs": 1,  # also for iql
+        "policy_update_delay": 1,
+        "buffer_size": 8,  # also for iql
+        "batch_size": 1,
+        # iql
+        "min_buffer_size": 4,
+        "sample_batch_size": 1,
+        "sample_sequence_length": 1,
```
Comment (RuanJohn) [inline test/conftest.py]: Just so it plays nice with `rec_iql` in the tests. 

```suggestion
        "sample_sequence_length": 2,
```

**Review by RuanJohn — CHANGES_REQUESTED** [review]
Just the update to the sample sequence length in the testing config for iql please.

File: test/conftest.py (hunk 1 above)
Comment (sash-a) [reply test/conftest.py]: switched back

File: test/conftest.py [hunk 6]
```diff
@@ -12,37 +12,41 @@
...
-from typing import Dict
+from typing import Dict, List, TypeAlias
+from test.utils import ConfigValue
 
 import pytest
 
 
 @pytest.fixture
-def fast_config() -> Dict[str, Dict[str, bool | int | float]]:
+def fast_config() -> Dict[str, ConfigValue]:
```
Comment (WiemKhlifi) [inline test/conftest.py:23]: I like this packing 😅

**Review by RuanJohn — APPROVED** [review]
:ship:

**Review by WiemKhlifi — APPROVED** [review]
Thanks @sash-a 🥳

**Pull request merged**